container.go

```diff
@@ -51,6 +51,15 @@ func NewContainerFromInspectOutput(output []byte) (Container, error) {
 	return container, nil
 }
 
-func (c Container) HostPort() string {
-	return c.Ports[c.Env["PORT"]]
+// Returns all of the container's ports
+func (c Container) ContainerPorts() []string {
+	ports := make([]string, 0, len(c.Ports))
+	for port := range c.Ports {
+		ports = append(ports, port)
+	}
+	return ports
+}
```

Comment:
What's the use-case for `ContainerPorts`?

Reply:
I should have added comments to describe my thought process here, sorry about that.

All of the changed code in `container.go` is to accommodate for how `matchers/be_available.go` works. 
In the matcher code we used to do 
```
response, err := http.Get(fmt.Sprintf("http://localhost:%s", container.HostPort()))
```
and this `container.HostPort()` command just did a lookup in the port map using whatever the `$PORT` environment variable was, as you can see in deleted line 54/55 in this file. Since we no longer use the `$PORT` environment variable in our port mappings, we can't do this. 
I added `ContainerPorts` as a way to loop through the map of the possible ports that have been set. We use this and `HostPort` in `matchers/be_available.go` solely to grab the first port, to use  to lookup the host port.

Since we need a port to look up the host port now, this seemed like a decent solution. The alternative I thought of would be to pass a port into the `BeAvailableMatcher`, and use that for the host port lookup.

Reply:
I have some concerns with this code as its return value is non-deterministic. This is a problem because the `BeAvailable` matcher makes an assumption that the port we want to probe is the first one in the list returned by `ContainerPorts`, and if that list returns randomly ordered ports, then that value cannot be reliably used.

Say I have 3 ports in the `Ports` map like this:
```go
c.Ports = map[string]string{
    "1111": "1234",
    "2222": "2345",
    "3333": "3456",
}
```

Then multiple calls to `ContainerPorts` could return the following results:
```
[ "1111", "2222", "3333"  ]
[ "2222", "3333", "1111"  ]
[ "1111", "3333", "2222"  ]
[ "3333", "2222", "1111"  ]
```

This is caused by how Go implements `range` over a map. According to the language spec [definition of the `range` clause](https://golang.org/ref/spec#RangeClause), the "iteration order over maps is not specified and is not guaranteed to be the same from one iteration to the next". This means that the array, unless it is sorted after it is created, will contain randomly ordered elements.

If the array contains more than one port, then it is essentially randomly ordered. Therefore, the port that is checked by `BeAvailable` will be random.

You could sort the array before returning it. This would give you a stable list of ports every time the `ContainerPorts` method is called. However, it might make sense to remove this method and push this logic directly into the `BeAvailable` matcher. The `Ports` field on a `Container` is exported and therefore can be accessed from inside the `BeAvailable` function. If we don't add a new method to the `Container`, then we keep the package API smaller and reduce the overall maintenance burden of the library. I'd say that feels like a good tradeoff for some minor added complexity in the `BeAvailable` matcher.

What do you think?

Reply:
@ryanmoran thank you for the explanation. I realized that `ContainerPorts` returns different orderings of ports, but I came to the conclusion that it wouldn't matter which port gets used in the `BeAvailableMatcher`, since any of the returned ports will be mapped to a host port and should be reachable by the `GET` request. This achieved the goal of checking the the container is available regardless of which container is chosen. Does that seem logical to you? Or do you still think that they should be sorted? Even if they were sorted, I'm not sure that just returning the first one off the list makes much of a difference.

I do agree that this would be better moved into `BeAvailable`, so I will definitely make that change